### DISCUSSION TITLE
Rest Request JSON File nested

### DISCUSSION THREAD

**Discussion by Sagestum** [Budibase/budibase#10488] [Help] [answered]
Hello everyone,

I'm having an issue with a REST request that returns nested JSON data. I've been trying to split the data using a "Transformer", but it's not working.

Here's an example of the JSON data (with random data placeholders):

<details><summary>Details</summary>
<p>

{
  "objects": {
    "lnkhalloTotschuess::27559": {
      "code": 0,
      "message": "",
      "class": "lnkhalloTotschuess",
      "key": "27559",
      "fields": {
        "hallo_id": "2",
        "hallo_name": "File1",
        "tschuess_id": "100377",
        "tschuess_name": "Dinge1",
        "tschuess_org_code": "BBB",
        "tschuess_id_finalclass_recall": "Bla1"
      }
    },
    "lnkhalloTotschuess::24908": {
      "code": 0,
      "message": "",
      "class": "lnkhalloTotschuess",
      "key": "24908",
      "fields": {
        "hallo_id": "1",
        "hallo_name": "File2",
        "tschuess_id": "100387",
        "tschuess_name": "Dinge2",
        "tschuess_org_code": "AAA",
        "tschuess_id_finalclass_recall": "Bla2"
      }
    }
  },
  "code": 0,
  "message": "Found: 11569"
}

</p>
</details> 

And here's the query I've been using with the transformer:

```javascript
function transform(data) {
  const result = [];
  const objects = data.objects;
  for (const key in objects) {
    const fields = objects[key].fields;
    result.push({
      tschuess_name: fields.tschuess_name,
      tschuess_org_code: fields.tschuess_org_code,
    });
  }
  return { results: result };
}


``` 

Unfortunately, this isn't returning any data. Can anyone help me figure out what I'm doing wrong? Thanks in advance!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Sagestum,

The main issue here I believe is with your transformer. You don't need to wrap this in a function. Where you are returning results at the end also doesn't need to be done in this way, just return the results.

I've attached screenshots of the whole process from start to finish showing various tabs of the rest area with your rest data.

**Fixed transformer**
<img width="963" alt="Fixed transformer" src="https://user-images.githubusercontent.com/126772285/236402638-12a1b01b-40c1-4ec5-9a42-81df59cf741d.png">

**Response**
<img width="975" alt="JSON response" src="https://user-images.githubusercontent.com/126772285/236402814-6c80856a-facb-4e6a-8d70-d03613dfeb10.png">

**Schema**
<img width="969" alt="schema creation" src="https://user-images.githubusercontent.com/126772285/236402878-979b0d3c-599b-47ed-9535-eb1774d3b678.png">

**Preview**
<img width="983" alt="preview" src="https://user-images.githubusercontent.com/126772285/236403019-f48c9b44-d912-488c-890a-97bf16de19ad.png">

**Result**
<img width="1467" alt="result" src="https://user-images.githubusercontent.com/126772285/236403058-d825fa03-588e-4b38-9dee-352ac34e6d7b.png">

This should hopefully fix your issues!

**Comment by Sagestum** [comment]
Thanks, that helped a log :D

**Comment by vikinim** [comment]
hi , 
i have tried this code for transformer .
but error " request did not return any data "

my json external api response is - 


{
    "status": "success",
    "data": {
        "NSE_EQ:NHPC": {
            "last_price": 78.93,
            "instrument_token": "NSE_EQ|INE848E01016"
        },
        "NSE_EQ:ABREL": {
            "last_price": 2816.1,
            "instrument_token": "NSE_EQ|INE055A01016"
        },
        "NSE_EQ:1003ISFL28": {
            "last_price": 985.85,
            "instrument_token": "NSE_EQ|INE413U07269"
        },
        "NSE_EQ:GS220230C": {
            "last_price": 0.0,
            "instrument_token": "NSE_EQ|IN000230C028"
        },
        "NSE_EQ:364D071124": {
            "last_price": 98.0,
            "instrument_token": "NSE_EQ|IN002023Z349"
        },
        "NSE_EQ:RENUKA": {
            "last_price": 41.43,
            "instrument_token": "NSE_EQ|INE087H01022"
        },
        "NSE_EQ:SABEVENTS": {
            "last_price": 16.25,
            "instrument_token": "NSE_EQ|INE860T01019"
        }
    }
}
 and i have used -
 
 
<img width="486" alt="Screenshot 2024-10-23 at 4 58 05 PM" src="https://github.com/user-attachments/assets/1cfa00c1-4e3b-4c78-b4d8-cefb53e725af">

and error is . 
<img width="486" alt="Screenshot 2024-10-23 at 4 59 02 PM" src="https://github.com/user-attachments/assets/93e7b15d-c8e9-41f7-9013-a9ffcea8dcb2">

raw data is fine . please guide.
